packages/back-end/src/util/organization.util.ts

```diff
@@ -68,27 +73,146 @@ export function roleToPermissionMap(
   return permissionsObj;
 }
 
-export function getUserPermissions(
+function getMergedPermissions(
+  existingPermissions: UserPermission,
+  existingRole: MemberRole | undefined,
+  teamInfo: TeamInterface | ProjectMemberRole,
+  org: OrganizationInterface
+): UserPermission {
+  const newPermissions = roleToPermissionMap(teamInfo.role, org);
+
+  if (!existingPermissions) {
+    // If there are no existingPermissions just return the permissions associated with the teamInfo
+    return {
+      environments: teamInfo.environments,
+      limitAccessByEnvironment: teamInfo.limitAccessByEnvironment,
+      permissions: roleToPermissionMap(teamInfo.role, org),
+    };
+  }
+
+  // Otherwise, we need to merge the existingPermissions with the newPermissions
+  for (const newPermission in newPermissions) {
+    if (
+      !existingPermissions.permissions[newPermission as Permission] &&
+      newPermissions[newPermission as Permission]
+    ) {
+      existingPermissions.permissions[newPermission as Permission] =
```

Comment:
Make a clone instead of mutating

Reply:
Why is `as` needed here?

Reply:
Without the `as` assertion, this is throwing the following error.

"Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'Partial<Record<Permission, boolean>>'.
  No index signature with a parameter of type 'string' was found on type 'Partial<Record<Permission, boolean>>'.ts(7053)"
  
  I've updated this slightly [here](https://github.com/growthbook/growthbook/pull/1640/commits/ce7c333a49e7324e14b42f8990eed2733fdc51ca) with a `isPermission` check to ensure that the `permission` string is as actual permission.
  
  It still isn't great. Going to take a break and come back to this as I've been trying to think of a cleaner solution for 30 minutes without success.